examples/common/python/connectors/direct/tcs_listener/tcs_workorder_receipt_handler.py

```diff
@@ -35,311 +39,399 @@ def __init__(self, kv_helper):
-                WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE, jrpc_id,
-                "Work order receipt already exist in the database. Hence invalid parameter")
+            wo_receipt = self.kv_helper.get("wo-receipts", wo_id)
+            if wo_receipt is None:
+                status, err_msg = self.__validate_work_order_receipt_create_req(
+                    input_value, wo_request
+                )
+                if status == True:
+                    self.kv_helper.set("wo-receipts", wo_id, input_json_str)
+                    raise JSONRPCDispatchException(
+                        JRPCErrorCodes.SUCCESS,
+                        "Receipt created successfully"
+                    )
+                else:
+                    raise JSONRPCDispatchException(
+                        JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                        err_msg
+                    )
+            else:
+                raise JSONRPCDispatchException(
+                    JRPCErrorCodes.INVALID_PARAMETER_FORMAT_OR_VALUE,
+                    "Work order receipt is already exists. Hence invalid parameter"
+                )
+
+# ------------------------------------------------------------------------------------------------
+
+    def __validate_work_order_receipt_create_req(self, wo_receipt_req, wo_request):
+        """
+        Function to validate the work order receipt create request parameters
+        Parameters:
+            - wo_receipt_req is work order receipt request as dictionary
+            - wo_request is string containing jrpc work order request
+        Returns - tuple containing validation status(Boolean) and error message(string)
+        """
+        # Valid parameters list
+        valid_params = ["workOrderId", "workerServiceId", "workerId", "requesterId",
+                        "receiptCreateStatus", "workOrderRequestHash", "requesterGeneratedNonce",
+                        "requesterSignature", "signatureRules","receiptVerificationKey"]
+        for key in wo_receipt_req["params"]:
+            if key not in valid_params:
+                return False, "Missing parameter " + key + " in the request"
+            else:
+                if key in ["workOrderId", "workerServiceId", "workerId",
+                           "requesterId", "requesterGeneratedNonce"]:
+                    if not is_valid_hex_str(wo_receipt_req["params"][key]):
+                        return False, "invalid data parameter for " + key
+                elif key in ["workOrderRequestHash", "requesterSignature"]:
+                    try:
+                        base64.b64decode(wo_receipt_req["params"][key])
+                    except Exception as e:
+                        return False, \
+                            "Invalid data format for " + key
+        receipt_type = wo_receipt_req["params"]["receiptCreateStatus"]
+        try:
+            receipt_enum_type = ReceiptCreateStatus(receipt_type)
```

Comment:
Can you remove receipt_enum_type variable since its not being used?

Reply:
Added it's usage.